src/glide/browser/base/content/test/motions/browser_words.ts

```diff
@@ -100,6 +100,46 @@ add_task(async function test_normal_diw() {
   });
 });
 
+add_task(async function test_normal_dw() {
+  await BrowserTestUtils.withNewTab(INPUT_TEST_FILE, async browser => {
+    const { set_text, test_edit, set_selection } = GlideTestUtils.make_input_test_helpers(browser, {
+      text_start: 1,
+    });
+
+    await set_text("Hello world", "dw at start of word deletes word + following space");
+    await set_selection(0);
+    await test_edit("dw", "world", 0, "w");
+
+    await set_text("Hello world", "dw from inside a word deletes to next word boundary (keeps preceding chars)");
+    await set_selection(2);
+    await test_edit("dw", "Heworld", 2, "w");
+
+    await set_text("Hello   world", "dw at start deletes word + all following spaces");
+    await set_selection(0);
+    await test_edit("dw", "world", 0, "w");
+
+    await set_text("Hello   world", "dw from inside a word deletes to next word, skipping extra spaces");
+    await set_selection(2);
+    await test_edit("dw", "Heworld", 2, "w");
+
+    await set_text("Hello, world", "dw stops at punctuation (keeps punctuation and following space)");
+    await set_selection(0);
+    await test_edit("dw", ", world", 0, ",");
+
+    await set_text("hello\nworld", "dw treats newline as whitespace and deletes it with the word");
+    await set_selection(0);
+    await test_edit("dw", "world", 0, "w");
+
+    await set_text("hello?.world", "dw on punctuation run deletes punctuation up to next word");
+    await set_selection(5);
+    await test_edit("dw", "helloworld", 5, "w");
+
+    await set_text("h h h", "dw deletes a single-letter word + following space");
+    await set_selection(2);
+    await test_edit("dw", "h h", 2, "h");
```

Comment:
nit: you can also set the expected char to make the test easier to read, and I also like to avoid duplicate chars for the same reason
```suggestion
    await set_selection(5, "?");
    await test_edit("dw", "helloworld", 5, "w");

    await set_text("h j k", "dw deletes a single-letter word + following space");
    await set_selection(2, "j");
    await test_edit("dw", "h k", 2, "k");
```